dasharo-compatibility/network-speed.robot

```diff
@@ -0,0 +1,134 @@
+# stuff in all test cases
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+
+# TODO:
+# - document which setup/teardown keywords to use and what are they doing
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+...                     AND
+...                     Skip If    not ${NIC_LOOP_SUPPORT}    NIC Loop not installed
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Variables ***
+${SCRIPT_TEXT}=     \#/bin/bash${ENTER}
+...                 ${BACKSPACE}${ENTER}
+...                 ${BACKSPACE}\#echo $1${ENTER}
+...                 ${BACKSPACE}\#echo $2${ENTER}
+...                 ${BACKSPACE}${ENTER}
+...                 ${BACKSPACE}\# Use ip -d link list to see maxmtu value of interface${ENTER}
+...                 ${BACKSPACE}\# maximum transmission unit (MTU) is a measurement representing the
+...                 ${BACKSPACE}largest data packet that a network-connected device will accept.${ENTER}
+...                 ${BACKSPACE}\#ip link set $1 mtu 9000${ENTER}
+...                 ${BACKSPACE}\#ip link set $2 mtu 9000${ENTER}
+...                 ${BACKSPACE}\# ----${ENTER}
+...                 ${BACKSPACE}${ENTER}
+...                 ${BACKSPACE}${ENTER}
+...                 ${BACKSPACE}\#echo Thanks${ENTER}
+...                 ${BACKSPACE}ip netns add ns_server${ENTER}
+...                 ${BACKSPACE}ip netns add ns_client${ENTER}
+...                 ${BACKSPACE}ip link set $1 netns ns_server${ENTER}
+...                 ${BACKSPACE}ip link set $2 netns ns_client${ENTER}
+...                 ${BACKSPACE}ip netns exec ns_server ip addr add dev $1 10.1.1.1/24${ENTER}
+...                 ${BACKSPACE}ip netns exec ns_client ip addr add dev $2 10.1.1.2/24${ENTER}
+...                 ${BACKSPACE}ip netns exec ns_server ip link set dev $1 up${ENTER}
+...                 ${BACKSPACE}ip netns exec ns_client ip link set dev $2 up${ENTER}
+...                 ${BACKSPACE}ip netns exec ns_server iperf3 -s &${ENTER}
+...                 ${BACKSPACE}ip netns exec ns_client iperf3 -c 10.1.1.1${ENTER}
+...                 ${BACKSPACE}${ENTER}
+...                 ${BACKSPACE}killall iperf3${ENTER}
+...                 ${BACKSPACE}${ENTER}
+...                 ${BACKSPACE}ip netns exec ns_client iperf3 -s &${ENTER}
+...                 ${BACKSPACE}ip netns exec ns_server iperf3 -c 10.1.1.2${ENTER}
+...                 ${BACKSPACE}${ENTER}
+...                 ${BACKSPACE}killall iperf3${ENTER}
+...                 ${BACKSPACE}${ENTER}
+...                 ${BACKSPACE}ip netns del ns_server${ENTER}
+...                 ${BACKSPACE}ip netns del ns_client
```

Comment:
It would really be better to use `Execute Command In Terminal`. Otherwise we may have less information in the test log if something goes wrong

Reply:
I was thinking that if for some reason this script would disconnect the DUT from the internet - this script would bring it back online. I think that running such test should be executed on the DUT side.

Reply:
Reworked as per suggestion from @miczyg1

Reply:
You have to make sure this test is not run on network interfaces providing connectivity with external world...

If it breaks the connection, then something is either wrong in the test or in the ionfrastracture